### PR TITLE
EAGLE-652:add topology health check to app framework

### PR DESCRIPTION
**Pull request by puyulu**
add topology health check to app framework

### CODE REVIEW

File: eagle-topology-check/eagle-topology-app/src/main/resources/META-INF/providers/org.apache.eagle.topology.TopologyCheckAppProvider.xml [hunk 1]
```diff
@@ -22,7 +22,6 @@
     <name>Topology Health Check</name>
     <version>0.5.0-incubating</version>
     <appClass>org.apache.eagle.topology.TopologyCheckApp</appClass>
```
Comment (haoch) [inline eagle-topology-check/eagle-topology-app/src/main/resources/META-INF/providers/org.apache.eagle.topology.TopologyCheckAppProvider.xml]: `appClass` is useless as well

File: eagle-topology-check/eagle-topology-app/src/main/java/org/apache/eagle/topology/extractor/hdfs/HdfsTopologyEntityParser.java [hunk 2]
```diff
@@ -220,7 +220,7 @@ private void createAllDataNodeEntities(JMXBean bean, long updateTime, TopologyEn
             final String hostname = liveNodes.getString(i);
             final JSONObject liveNode = jsonNodesObject.getJSONObject(hostname);
 
-            HdfsServiceTopologyAPIEntity entity = createHdfsServiceEntity(TopologyConstants.DATA_NODE_ROLE, hostname, updateTime);
+            HdfsServiceTopologyAPIEntity entity = createHdfsServiceEntity(TopologyConstants.DATA_NODE_ROLE, EntityBuilderHelper.getValidHostName(hostname), updateTime);
             final Number configuredCapacity = (Number) liveNode.get(DATA_NODE_CAPACITY);
             entity.setConfiguredCapacityTB(Double.toString(configuredCapacity.doubleValue() / 1024.0 / 1024.0 / 1024.0 / 1024.0));
```
Comment (haoch) [inline eagle-topology-check/eagle-topology-app/src/main/java/org/apache/eagle/topology/extractor/hdfs/HdfsTopologyEntityParser.java]: Use static final value `Long TB = 1024.0*1024.0*1024.0*1024.0` then use 

~~~
entity.setConfiguredCapacityTB(Double.toString(configuredCapacity.doubleValue() / TB));
~~~

**Comment by haoch** [comment]
LGTM if you could fix above two small issues.

File: eagle-topology-check/eagle-topology-app/src/main/resources/META-INF/providers/org.apache.eagle.topology.TopologyCheckAppProvider.xml (hunk 1 above)
Comment (puyulu) [inline eagle-topology-check/eagle-topology-app/src/main/resources/META-INF/providers/org.apache.eagle.topology.TopologyCheckAppProvider.xml]: removed unuse class name

File: eagle-topology-check/eagle-topology-app/src/main/java/org/apache/eagle/topology/extractor/hdfs/HdfsTopologyEntityParser.java (hunk 2 above)
Comment (puyulu) [inline eagle-topology-check/eagle-topology-app/src/main/java/org/apache/eagle/topology/extractor/hdfs/HdfsTopologyEntityParser.java]: changed the hard code to final value

**Comment by puyulu** [comment]
removed the unused code after code review

**Comment by haoch** [comment]
LGTM, thanks for the fix.

**Pull request closed without merging**